src/utils/fees.ts

```diff
@@ -46,46 +60,132 @@ function isEIP1559Fees(chain: ChainId) {
-        network: store.state.activeNetwork,
-        walletId: store.state.activeWalletId,
-        asset: asset,
-        accountId: accountId
-      }) as Client<
-        BitcoinEsploraApiProvider,
-        BitcoinBaseWalletProvider
-      >;
-      const feePerBytes = Object.values(assetFees).map((fee) => fee.fee)
-      const value = getMax ? undefined : currencyToUnit(cryptoassets[asset], amount)
-      try {
-        const txs = feePerBytes.map((fee) => ({ value, fee }))
-
-        const totalFees = await client.wallet.getTotalFees(txs, getMax)
-        for (const [speed, fee] of Object.entries(assetFees)) {
-          const totalFee = unitToCurrency(cryptoassets[asset], totalFees[fee.fee])
-          sendFees[speed as keyof FeeDetails] = totalFee
-        }
-      } catch (e) {
-        console.error(e)
-      }
+/*
+ * Calculates fee for EIP1559 or returns fee for non EIP1559 chains per gas or byte (unit)
+ */
+function feePerUnit(suggestedGasFee: FeeType, chain: ChainId): number {
+  if (suggestedGasFee === undefined || chain === undefined) {
+    throw new Error('feePerUnit: Incorrect input!');
+  }
+
+  /*
+   * In case it is EIP1559 chain then `maxFeePerGas` will be set.
+   * Otherwise its non EIP1559 (this includes both EVM and Non EVM chains) and calculations are done based on `fee`.
+   */
+  if (isEIP1559Fees(chain) && (<EIP1559Fee>suggestedGasFee).maxFeePerGas) {
+    return maxFeePerUnitEIP1559(<EIP1559Fee>suggestedGasFee);
+  }
+
+  return suggestedGasFee as number;
+}
+
+async function getSendTxFees(accountId: AccountId, asset: Asset, amount?: BN) {
+  const assetChain = cryptoassets[asset]?.chain;
+  if (!assetChain) {
+    throw new Error(`getSendFeeEstimations: asset chain not available for ${asset}`);
+  }
+
+  const feeAsset = getFeeAsset(asset) || getNativeAsset(asset);
+  if (!feeAsset) {
+    throw new Error(`getSendFeeEstimations: fee asset not available for ${asset}`);
+  }
+
+  const suggestedGasFees = store.getters.getSuggestedFeePrices(asset);
+  if (!suggestedGasFees) {
+    throw new Error(`getSendFeeEstimations: fees not avaibale for ${asset}`);
+  }
+
+  if (assetChain === ChainId.Bitcoin) {
+    return sendTxFeesInBTC(accountId, feeAsset, suggestedGasFees, amount);
+  } else {
+    return sendTxFeesInNativeAsset(feeAsset, suggestedGasFees);
+  }
+}
+
+/*
+ * Send fee estimation method for all EIP1559 and non EIP1559 chains
+ */
+function sendTxFeesInNativeAsset(feeAsset: Asset, suggestedGasFees: FeeDetails, sendFees?: SendFees) {
+  const assetChain = cryptoassets[feeAsset]?.chain;
+  const _sendFees = sendFees ?? newSendFees();
+
+  for (const [speed, fee] of Object.entries(suggestedGasFees)) {
+    const _speed = speed as keyof FeeDetails;
+
+    const _fee: number = feePerUnit(fee.fee, assetChain);
+
+    _sendFees[_speed] = _sendFees[_speed].plus(getSendFee(feeAsset, _fee));
+  }
+
+  return _sendFees;
+}
+
+/*
+ * Send fee estimation method for BTC
+ */
+async function sendTxFeesInBTC(
```

Comment:
The name of this function is focusing on the unit returned, `InBTC` This gives the impression that there are other function returning values in different units 
IMO better name would be `sendBitcoinTxFees`

Reply:
`sendTxFeesInBTC` -> `sendBitcoinTxFees`
But then I have to rename: `sendTxFeesInNativeAsset` to have consistent names for all methods?
Any proposal?